logdna/resource_view.go

```diff
@@ -99,18 +103,16 @@ func buildChannels(emailchannels []interface{}, pagerdutychannels []interface{},
 		url := i["url"].(string)
 
 		headersMap := make(map[string]string)
-		templateMap := make(map[string]string)
 
 		for k, v := range headers {
 			headersMap[k] = v.(string)
 		}
 
-		for k, v := range bodytemplate {
-			templateMap[k] = v.(string)
-		}
+		var bt map[string]interface{}
+		err = json.Unmarshal([]byte(bodytemplate), &bt)
```

Comment:
just define `err` here (no need to do it on line 37)

also this should just check and return the error immediately/ otherwise the loop is continuing and `err` gets overwritten each time

```golang
err := json.Unmarshal([]byte(bodytemplate), &bt)
if err {
  return myChannels, err
}
```